src/qibolab/platforms/abstract.py

```diff
@@ -38,45 +118,27 @@ def __init__(self, name, runcard):
             self.resonator_type = "3D"
         else:
             self.resonator_type = "2D"
-        self.hardware_avg = self.settings["settings"]["hardware_avg"]
-        self.sampling_rate = self.settings["settings"]["sampling_rate"]
-        self.repetition_duration = self.settings["settings"]["repetition_duration"]
 
-        self.qubits = self.settings["qubits"]
         self.topology = self.settings["topology"]
         self.channels = self.settings["channels"]
-        self.qubit_channel_map = self.settings["qubit_channel_map"]
-
-        # Load Characterization settings
-        self.characterization = self.settings["characterization"]
-        # Load Native Gates
-        self.native_gates = self.settings["native_gates"]
 
         self.instruments = {}
         # Instantiate instruments
-        for name in self.settings["instruments"]:
-            lib = self.settings["instruments"][name]["lib"]
-            i_class = self.settings["instruments"][name]["class"]
-            address = self.settings["instruments"][name]["address"]
+        for name, inst_settings in self.settings.get("instruments", {}).items():
+            lib = inst_settings["lib"]
+            i_class = inst_settings["class"]
+            address = inst_settings["address"]
             from importlib import import_module
 
             InstrumentClass = getattr(import_module(f"qibolab.instruments.{lib}"), i_class)
             instance = InstrumentClass(name, address)
             self.instruments[name] = instance
 
-        # Generate qubit_instrument_map from qubit_channel_map and the instruments' channel_port_maps
-        self.qubit_instrument_map = {}
-        for qubit in self.qubit_channel_map:
-            self.qubit_instrument_map[qubit] = [None, None, None]
-            for name in self.instruments:
-                if "channel_port_map" in self.settings["instruments"][name]["settings"]:
-                    for channel in self.settings["instruments"][name]["settings"]["channel_port_map"]:
-                        if channel in self.qubit_channel_map[qubit]:
-                            self.qubit_instrument_map[qubit][self.qubit_channel_map[qubit].index(channel)] = name
-                if "s4g_modules" in self.settings["instruments"][name]["settings"]:
-                    for channel in self.settings["instruments"][name]["settings"]["s4g_modules"]:
-                        if channel in self.qubit_channel_map[qubit]:
-                            self.qubit_instrument_map[qubit][self.qubit_channel_map[qubit].index(channel)] = name
+        # Instantiate Qubit objects and map them to channels and instruments
+        self.qubits = [Qubit(q, self.settings) for q in self.settings["qubits"]]
+
+        # Load the rest of settings
+        self.reload_settings()
```

Comment:
This does not read well, we cannot call `reload_settings()` in the init, as we never loaded the settings before.
I would rename this function to `load_settings_that_dont_require_connection()` or something shorter :) and then have both `init()` and `reload_settings()` call it.